File: deps/v8/test/debugging/wasm/gdb-server/test_files/test_float.js
// Copyright 2020 the V8 project authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

load('test/mjsunit/wasm/wasm-module-builder.js');

var builder = new WasmModuleBuilder();
builder
    .addFunction('mul', kSig_f_ff)
    // input is 2 args of type float and output is float
    .addBody([
      kExprLocalGet, 0,  // local.get f0
      kExprLocalGet, 1,  // local.get f1
      kExprF32Mul,       // f32.mul i0 i1
    ])
    .exportFunc();

const instance = builder.instantiate();
const wasm_f = instance.exports.mul;

function f() {
  var result = wasm_f(12.0, 3.5);
  return result;
}

try {
  let val = f();
  print('float result: ' + val);
} catch (e) {
  print('*exception:* ' + e);
}

File: deps/v8/test/debugging/wasm/gdb-server/test_files/test_float.py
# Copyright 2020 the V8 project authors. All rights reserved.
# Use of this source code is governed by a BSD-style license that can be
# found in the LICENSE file.

# 0x00 (module
# 0x08   [type]
#        (type $type0 (func (param f32 f32) (result f32)))
# 0x11   [function]
# 0x15   (export "mul" (func $func0))
# 0x1e   [code]
# 0x20   (func $func0 (param $var0 f32) (param $var1 f32) (result f32)
# 0x23     get_local $var0
# 0x25     get_local $var1
# 0x27     f32.mul
#   )
# )
# 0x29   [name]

ARG_0 = 12.0
ARG_1 = 3.5
FUNC_START_ADDR = 0x23
